plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/writer/TransactionLogSynchronizerManager.java

```diff
@@ -43,8 +42,8 @@ public TransactionLogSynchronizerManager(
 
     public TransactionLogSynchronizer getSynchronizer(String tableLocation)
     {
-        String uriScheme = new Path(tableLocation).toUri().getScheme();
-        checkArgument(uriScheme != null, "URI scheme undefined for " + tableLocation);
+        String uriScheme = Location.of(tableLocation).scheme()
```

Comment:
when should `Location.of(...).scheme()` be used
and when `new URI(...).getScheme()` is preferred?

Reply:
The former must always be used. Locations are not valid URIs, since they can contain unescaped `%` characters, for example. The parsing behavior is more similar to Hadoop `new Path(x)` rather than `URI.create(x)`.

The `toUriFormat()` method uses the URI constructor with the `path` argument, which is the actual path value and not the URI-escaped path (it won't be parsed or un-escaped).

Reply:
I think i more or less understand that, but maybe some examples would help.

Anyway, does `toUriFormat` work because Delta paths are relative?
Does this function need an update in https://github.com/trinodb/trino/pull/17038?

cc @findinpath